ephios/core/services/notifications/backends.py

```diff
@@ -25,25 +30,55 @@ def enabled_notification_backends():
         yield from (b() for b in backends)
 
 
+def send_a_notification(backend):
+    with transaction.atomic():
+        # grab an untried notification and mark it
+        if (
+            notification := Notification.objects.exclude(backends_tried__has_key=backend.slug)
+            .exclude(backends_tried__has_key=backend.slug)
+            .first()
+        ) is None:
+            return
```

Comment:
Now this should only be None if another thread has snacked Notifications away, but if, we uselessly lock the database here.
Also, is there a nicer way in django to this "get me one and mark it as used" pattern?

Maybe we can use [`update`](https://docs.djangoproject.com/en/dev/ref/models/querysets/#update) with a limit=1?

Reply:
not sure if the update variant makes it more elegant as you'd need to make sure that you hit the right object within the limit=1

Reply:
do I miss smth or are the two exclude() calls exactly identical?